scripts/apps/dashboard/user-activity/components/UserActivityWidget.tsx

```diff
@@ -0,0 +1,411 @@
+                });
+            },
+        },
+        {
+            id: 'moved',
+            label: gettext('Moved to a working stage by this user'),
+            query(fetchFn) {
+                return fetchFn('archive_history', {...getQueryMovedByUser(userId)})
+                    .then((res) => {
+                        res._items = res._items.filter(onlyHistoryItemsInWorkingStage(services));
+
+                        return res;
+                    });
+            },
+        },
+    ];
+};
+
+function onlyHistoryItemsInWorkingStage({desks}) {
+    return (historyItem) => {
+        const stage = getStageForItem(historyItem, {desks});
+
+        return stage.working_stage === true;
+    };
+}
+
+function filterOutItemsInIncomingStage({desks}) {
+    return (item) => {
+        const stage = getStageForItem(item, {desks});
+
+        return stage.default_incoming === false;
+    };
+}
+
+function getStageForItem(item, {desks}) {
+    const stageId = item.task?.stage;
+
+    if (!stageId) {
+        return false;
+    }
+
+    const stage = desks.stageLookup[stageId] || null;
+
+    if (!stage) {
+        console.warn('Tried to find a stage with an invalid id', stageId);
+    }
+
+    return stage;
+}
+
+export default class UserActivityWidget extends React.Component<{}, IState> {
+    services: any;
+
+    constructor(props) {
+        super(props);
+
+        this.services = {
+            $anchorScroll: ng.get('$anchorScroll'),
+            $location: ng.get('$location'),
+            $q: ng.get('$q'),
+            $rootScope: ng.get('$rootScope'),
+            $timeout: ng.get('$timeout'),
+            activityService: ng.get('activityService'),
+            api: ng.get('api'),
+            authoringWorkspace: ng.get(
+                'authoringWorkspace',
+            ) as AuthoringWorkspaceService,
+            cards: ng.get('cards'),
+            datetime: ng.get('datetime'),
+            desks: ng.get('desks'),
+            metadata: ng.get('metadata'),
+            search: ng.get('search'),
+            superdesk: ng.get('superdesk'),
+        };
+
+        this.state = {
+            userSearchField: '',
+            searchOpen: true,
+            groups: null,
+            groupsData: null,
+            loading: false,
+        };
+    }
+
+    componentDidMount() {
+        [
+            'item:lock',
+            'item:unlock',
+            'item:spike',
+            'item:move',
+            'item:publish',
+            'item:update',
+        ].map((event) =>
+            this.services.$rootScope.$on(event, this.refreshItems.bind(this)),
+        );
+    }
+
+    refreshItems() {
+        if (this.state.user) {
+            this.fetchGroupsData();
+        }
+    }
+
+    async fetchItems(group: IGroup) {
+        const {api, search} = this.services;
+        const {_items} = typeof group.query === 'function'
+            ? await group.query((repo, filters) => genericFetch({search, api}, repo, filters))
```

Comment:
I see you are trying to sneak in async/await again :smile: We aren't using async/await in the codebase yet and I think it's simpler to keep it this way. We could consider it if it added significant value like flattened a complex promise chain, but it's [almost the same in this case](https://user-images.githubusercontent.com/2076953/72738463-ea9aac00-3ba1-11ea-8753-ccf0f40f511c.png)

Reply:
I see where you are coming from and even half-agree, but in the context that we are using asynchronous code*, the difference is quite small, plus we are more familiar to promises so I'd stick to it until we bump into use cases where using fetch would make a significant positive difference.

\* async/await probably helps more on the server-side due to the nature of the domain